code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,918 @@
+
+*/
+
+/obj/item/slimecross/warping
+	name = "warped extract"
+	desc = "It just won't stay in place."
+	icon_state = "warping"
+	effect = "warping"
+	colour = "grey" ///default color of the crossbreed
+	var/obj/effect/warped_rune/runepath ///what runes will be drawn depending on the crossbreed color
+	var/warp_charge = 1 /// the number of "charge" a bluespace crossbreed start with
+	var/max_charge = 1 ///max number of charge, might be different depending on the crossbreed (all crossbreed have 1 max charge for now)
+	var/storing_time = 15 ///time it takes to store the rune back into the crossbreed
+	var/drawing_time = 15 ///time it takes to draw the rune
+
+/obj/effect/warped_rune
+	name = "warped rune"
+	desc = "An unstable rune born of the depths of bluespace"
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "greyspace_rune"
+	var/runepath = /obj/effect/warped_rune  ///serves as a way for some specific runes to identify themselves in the turf.TODO: you can just use src dumbass
+	var/storing_time = 5 ///is only used for bluespace crystal erasing as of now
+	anchored = TRUE
+	layer = MID_TURF_LAYER
+	resistance_flags = FIRE_PROOF  //It's only fireproof because of the fire rune and because how do you burn a rune anyway.
+	var/turf/T //used on a LOT of runes, so we might as well put it here
+	var/cooldown = 0 //cooldown for the rune process() only applies to certain runes
+	var/max_cooldown = 100
+	move_resist = INFINITY //here to avoid the rune being moved since it only sets it's turf once when it's drawn. doesn't include admin fuckery.
+
+///runes can also be deleted by bluespace crystals if the xenobiologist is fucking everyone up and you need a way to destroy the rune.
+/obj/effect/warped_rune/attackby(obj/item/stack/BC, mob/user)
+	if(!istype(BC,/obj/item/stack/sheet/bluespace_crystal) && !istype(BC,/obj/item/stack/ore/bluespace_crystal))
+		return
+	if(do_after(user, storing_time,target = src)) //the time it takes to nullify it depends on the rune too
+		to_chat(user, "<span class='notice'>You nullify the effects of the rune with the bluespace crystal!</span>")
+		qdel(src)
+		BC.amount--
+		playsound(src, 'sound/effects/phasein.ogg', 20, TRUE)
+		if(BC.amount <= 0)
+			qdel(BC)
+
+
+
+
+/obj/effect/warped_rune/acid_act()
+	visible_message("<span class='warning'>[src] has been dissolved by the acid</span>")
+	playsound(src, 'sound/items/welder.ogg', 150, TRUE)
+	qdel(src)
+
+///nearly all runes use their turf in some way so we set T to their turf automatically, the rune also start on cooldown if it uses one.
+/obj/effect/warped_rune/Initialize()
+	.=..()
+	T = get_turf(src)
+	RegisterSignal(T, COMSIG_COMPONENT_CLEAN_ACT, .proc/clean_rune)
+	cooldown = world.time + max_cooldown
+
+/obj/effect/warped_rune/proc/clean_rune()
+	qdel(src)
+
+ ///using the extract on the floor will "draw" the rune.
+/obj/item/slimecross/warping/afterattack(turf/target, mob/user, proximity)
```

Comment:
Keep the type of the input as the possible ones, not just the one you want.
Other coders might want to change the proc and this can be misleading.